src/main/java/core/basesyntax/data/dao/FileWriterImpl.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax.data.dao;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.util.Map;
+
+public class FileWriterImpl implements FileWriter {
+    @Override
+    public void writeFile(String fileName, Map<String, Integer> inventory) {
+        try (BufferedWriter bw = new BufferedWriter(new java.io.FileWriter(fileName))) {
+            bw.write("fruit,quantity");
+            bw.newLine();
+
+            for (Map.Entry<String, Integer> entry : inventory.entrySet()) {
+                bw.write(entry.getKey() + "," + entry.getValue());
+                bw.newLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Error writing to file " + fileName, e);
```

Comment:
Ensure the exception message is informative. Consider including more details about the error, such as the specific operation that failed, to help with debugging.